dynamodb.tf

```diff
@@ -0,0 +1,30 @@
+resource "aws_dynamodb_table" "geff_batch_locking_table" {
+  count        = var.create_dynamodb_table ? 1 : 0
+  name         = "${local.geff_prefix}_batch_locking_table"
+  billing_mode = "PAY_PER_REQUEST"
+  hash_key     = "batch_id"
+
+  attribute {
+    name = "batch_id"
+    type = "S"
+  }
+
+  ttl {
+    attribute_name = "TimeToExist"
+    enabled        = true
+  }
+
+  tags = {
+    name = "${local.geff_prefix}_batch_locking_table"
```

Comment:
Yeah, I think we should have general tagging on the provider and then overwrite/add at the resource level.  I started putting general tags in the provider for Anecdotes.  But I think we should agree on the format and what information we want to capture.  For a module like this, we can look into having an input map variable for users to pass in multiple tags.

```provider "aws" {
  region  = var.aws_region
  default_tags {
    tags = {
      environment = "prod"
      owner       = "compliance"
      project     = "anecdotes"
    }
  }
}```


But the existing resource naming in this module is using hyphen instead of underscore.  For this PR, keeping it consistent with the existing resource naming should be fine for now until we agree on the general tagging name and approach.
```resource "aws_api_gateway_rest_api" "ef_to_lambda" {
  name = "${local.geff_prefix}-api-gateway"

  endpoint_configuration {
    types = [
      "REGIONAL",
    ]
  }
}```